sky/execution.py

```diff
@@ -158,16 +159,32 @@ def _execute(
             cluster_name)
         cluster_exists = existing_handle is not None
 
+    stages = stages if stages is not None else list(Stage)
+
     backend = backend if backend is not None else backends.CloudVmRayBackend()
-    if not isinstance(backend, backends.CloudVmRayBackend
-                     ) and idle_minutes_to_autostop is not None:
+    if isinstance(backend, backends.CloudVmRayBackend):
+        if down and idle_minutes_to_autostop is None:
+            # Use autostop(down) to terminate the cluster after the task is
+            # done. Otherwise, the cluster will be immediately terminated when
+            # the user detach from the execution.
+            idle_minutes_to_autostop = 0
+        if idle_minutes_to_autostop is not None:
+            if idle_minutes_to_autostop == 0:
+                verb = 'torn down' if down else 'stopped'
+                logger.warning(f'{colorama.Fore.LIGHTBLACK_EX}Setting '
```

Comment:
How about printing this warning only for `-i 0` but not for `--down`? The latter is a "normal" use case so shouldn't see a warning.

Reply:
How about `logger.info` and say something like "The cluster will be {verb}-ed after 1 minute of idleness (after all jobs finish)."?

This is to (1) avoid "warning" for a normal use case, and (2) not show the reason, as it's hard for users to understand.

I still think not showing this for `--down` is better as the UX is simpler. But up to you!